integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/AlterTableRenameCommandVisitor.java

```diff
@@ -1,46 +1,38 @@
 package io.openlineage.spark.agent.lifecycle.plan;
 
-import static io.openlineage.spark.agent.util.PlanUtils.datasourceFacet;
-import static io.openlineage.spark.agent.util.PlanUtils.schemaFacet;
-
 import io.openlineage.client.OpenLineage;
 import io.openlineage.spark.agent.facets.PreviousTableNameFacet;
 import io.openlineage.spark.agent.util.DatasetIdentifier;
 import io.openlineage.spark.agent.util.PathUtils;
 import io.openlineage.spark.agent.util.PlanUtils;
+import io.openlineage.spark.api.DatasetFactory;
+import io.openlineage.spark.api.OpenLineageContext;
+import io.openlineage.spark.api.QueryPlanVisitor;
 import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
-import org.apache.spark.sql.SparkSession;
-import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
 import org.apache.spark.sql.catalyst.catalog.CatalogTable;
-import org.apache.spark.sql.catalyst.catalog.SessionCatalog;
 import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
 import org.apache.spark.sql.execution.command.AlterTableRenameCommand;
 
 @Slf4j
 public class AlterTableRenameCommandVisitor
-    extends QueryPlanVisitor<AlterTableRenameCommand, OpenLineage.Dataset> {
-
-  private final SparkSession sparkSession;
+    extends QueryPlanVisitor<AlterTableRenameCommand, OpenLineage.OutputDataset> {
 
-  public AlterTableRenameCommandVisitor(SparkSession sparkSession) {
-    this.sparkSession = sparkSession;
+  public AlterTableRenameCommandVisitor(OpenLineageContext openLineage) {
```

Comment:
_Minor_: I'd try and keep naming consistent. Looks like `context` is the winner?